core/include/detray/utils/ranges/detail/iterator_functions.hpp

```diff
@@ -55,6 +56,22 @@ inline constexpr bool random_access_iterator_v =
     std::is_base_of_v<detray::ranges::random_access_iterator_tag,
                       typename std::iterator_traits<I>::iterator_category>;
 
+// Iterator concepts
+template <class I>
+concept input_iterator = detray::ranges::input_iterator_v<I>;
+
+template <class I>
+concept output_iterator = detray::ranges::output_iterator_v<I>;
+
+template <class I>
+concept forward_iterator = detray::ranges::forward_iterator_v<I>;
+
+template <class I>
+concept bidirectional_iterator = detray::ranges::bidirectional_iterator_v<I>;
+
+template <class I>
+concept random_access_iterator = detray::ranges::random_access_iterator_v<I>;
```

Comment:
OK, let me see if I can remove those traits, because I don't really want the trait to result in different constraints than the concept, i.e.
```
template<typename T>
requires detray::ranges::forward_iterator_v<T>
```
and
```
template<detray::ranges::forward_iterator T>
```
should come out the same imo. I will make a proper cleanout at some point about which parts of detray ranges to keep and where to use std ranges, but I would like to do that in a separate PR if possible

Reply:
Done. I left the concepts as an alias though, until I know how I can replace/mix std;;ranges and detray::ranges